### PR TITLE
HPCC-32862 Add Copilot article to devdoc

### PR DESCRIPTION
**Pull request by JamesDeFabia**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by g-pan — APPROVED** [review]
reviewed and approved

File: devdoc/userdoc/copilot/CopilotPromptTips.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+- Write a step-by-step guide on how to [specific task or procedure]. Number the steps to improve clarity.
+
+  - **Example:** Write a step-by-step guide on how to make a PB & J sandwich. Number the steps to improve clarity.
+
+## Specific Prompts
+
+These prompts are more focused and are meant to accomplish a specific purpose. They are included here to spark your imagination.
+
+If you think of others that could be included here to share witht he world, please send your ideas to <docfeedback@hpccsystems.com>.
```
Comment (stuartort) [inline devdoc/userdoc/copilot/CopilotPromptTips.md:42]: spelling witht he -> with the

**Review by stuartort — CHANGES_REQUESTED** [review]
Your examples are good but simple 1 line prompts.  These are good when using the interactive mode in file.  Developers may find it easier to use the copilot chat inside of vscode to type long detailed instructions for code development.  This can be added either in this PR or create a jira (mention it here) and do it in another PR

In addition, you should probably discuss tips for saving prompts.  Once you use a prompt in a code window, it is gone.  In chat windows it may be saved but still a good idea to save it in another file (you never know when you get code generated, hand edit it and mess it up.  It is great when you have the prompt available to regenerate your code

File: devdoc/userdoc/copilot/CopilotPromptTips.md (hunk 1 above)
Comment (JamesDeFabia) [reply devdoc/userdoc/copilot/CopilotPromptTips.md:42]: Fixed the typo. Creating a Jira to enhance this article with tips to save prompt history and other enhacements

**Review by stuartort — APPROVED** [review]

**Pull request merged**